python/bind_data_access.h

```diff
@@ -259,7 +283,7 @@ template <class... Ts> struct as_VariableViewImpl {
         [&o](const auto &data) {
           data[0] = o.cast<typename std::decay_t<decltype(data)>::value_type>();
         },
-        get<get_variances>(view));
+        *get<get_variances>(view));
   }
 };
```

Comment:
The changes to this file make the logic quite hard to understand. I am counting **seven** places where you check for values/variances and then return none/optional. This does not look very maintainable, I think we should try and come up with a simpler solution.

Reply:
I agree, all stuff in this file hasn't been designed for all these checks, but now I would fix bugs first (also with `set` functionality), and then do some refactoring.